OneSignalSDK/onesignal/src/main/java/com/onesignal/ActivityLifecycleHandler.java

```diff
@@ -42,95 +43,53 @@
-            sKeyboardListeners.put(key, keyboardListener);
-        }
-        sSystemConditionObservers.put(key, systemConditionObserver);
-    }
+    private Activity curActivity = null;
+    private boolean nextResumeIsFirstActivity = false;
 
-    static void setActivityAvailableListener(String key, ActivityAvailableListener activityAvailableListener) {
-        sActivityAvailableListeners.put(key, activityAvailableListener);
-        if (curActivity != null)
-            activityAvailableListener.available(curActivity);
-    }
-
-    static void removeSystemConditionObserver(String key) {
-        sKeyboardListeners.remove(key);
-        sSystemConditionObservers.remove(key);
-    }
-
-    static void removeActivityAvailableListener(String key) {
-        sActivityAvailableListeners.remove(key);
-    }
-
-    private static void setCurActivity(Activity activity) {
-        curActivity = activity;
-        for (Map.Entry<String, ActivityAvailableListener> entry : sActivityAvailableListeners.entrySet()) {
-            entry.getValue().available(curActivity);
-        }
-
-        try {
-            ViewTreeObserver treeObserver = curActivity.getWindow().getDecorView().getViewTreeObserver();
-            for (Map.Entry<String, OSSystemConditionController.OSSystemConditionObserver> entry : sSystemConditionObservers.entrySet()) {
-                KeyboardListener keyboardListener = new KeyboardListener(entry.getValue(), entry.getKey());
-                treeObserver.addOnGlobalLayoutListener(keyboardListener);
-                sKeyboardListeners.put(entry.getKey(), keyboardListener);
-            }
-        } catch (RuntimeException e) {
-            // Related to Unity Issue #239 on Github
-            // https://github.com/OneSignal/OneSignal-Unity-SDK/issues/239
-            // RuntimeException at ActivityLifecycleHandler.setCurActivity on Android (Unity 2.9.0)
-            e.printStackTrace();
-        }
-    }
-
-    static void onConfigurationChanged(Configuration newConfig) {
+    void onConfigurationChanged(Configuration newConfig, Activity activity) {
         // If Activity contains the configChanges orientation flag, re-create the view this way
         if (curActivity != null && OSUtils.hasConfigChangeFlag(curActivity, ActivityInfo.CONFIG_ORIENTATION)) {
-            logOrientationChange(newConfig.orientation);
-            onOrientationChanged();
+            logOrientationChange(newConfig.orientation, activity);
+            onOrientationChanged(activity);
         }
     }
 
-    static void onActivityCreated(Activity activity) {
+    void onActivityCreated(Activity activity) {
     }
 
-    static void onActivityStarted(Activity activity) {
+    void onActivityStarted(Activity activity) {
     }
 
-    static void onActivityResumed(Activity activity) {
+    void onActivityResumed(Activity activity) {
+        OneSignal.Log(OneSignal.LOG_LEVEL.DEBUG, "onActivityResumed: " + activity);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.